src/mds/MDSMap.h

```diff
@@ -543,6 +543,7 @@ class MDSMap {
   static bool state_transition_valid(DaemonState prev, DaemonState next);
 
   CompatSet compat;
+  bool is_client = false;
```

Comment:
you should add a field to MMDSMap instead here. also add a parameter (with default value) to MDSMap::encode